samples/dnn/dasiamrpn_tracker.py

```diff
@@ -0,0 +1,311 @@
+    left_pad = int(max(0., -context_xmin))
+    top_pad = int(max(0., -context_ymin))
+    right_pad = int(max(0., context_xmax - im_sz[1] + 1))
+    bottom_pad = int(max(0., context_ymax - im_sz[0] + 1))
+
+    context_xmin = context_xmin + left_pad
+    context_xmax = context_xmax + left_pad
+    context_ymin = context_ymin + top_pad
+    context_ymax = context_ymax + top_pad
+    r, c, k = im.shape
+
+    if any([top_pad, bottom_pad, left_pad, right_pad]):
+        te_im = np.zeros((r + top_pad + bottom_pad, c + left_pad + right_pad, k), np.uint8)
+        te_im[top_pad:top_pad + r, left_pad:left_pad + c, :] = im
+        if top_pad:
+            te_im[0:top_pad, left_pad:left_pad + c, :] = avg_chans
+        if bottom_pad:
+            te_im[r + top_pad:, left_pad:left_pad + c, :] = avg_chans
+        if left_pad:
+            te_im[:, 0:left_pad, :] = avg_chans
+        if right_pad:
+            te_im[:, c + left_pad:, :] = avg_chans
+        im_patch_original = te_im[int(context_ymin):int(context_ymax + 1), int(context_xmin):int(context_xmax + 1), :]
+    else:
+        im_patch_original = im[int(context_ymin):int(context_ymax + 1), int(context_xmin):int(context_xmax + 1), :]
+
+    if not np.array_equal(model_sz, original_sz):
+        im_patch = cv.resize(im_patch_original, (model_sz, model_sz))
+    else:
+        im_patch = im_patch_original
+
+    return im_patch.astype(float)
+
+def generate_anchor(total_stride, scales, ratios, score_size):
+    anchor_num = len(ratios) * len(scales)
+    anchor = np.zeros((anchor_num, 4),  dtype=np.float32)
+    size = total_stride * total_stride
+    count = 0
+
+    for ratio in ratios:
+        ws = int(np.sqrt(size / ratio))
+        hs = int(ws * ratio)
+        for scale in scales:
+            wws = ws * scale
+            hhs = hs * scale
+            anchor[count, 0] = 0
+            anchor[count, 1] = 0
+            anchor[count, 2] = wws
+            anchor[count, 3] = hhs
+            count += 1
+
+    anchor = np.tile(anchor, score_size * score_size).reshape((-1, 4))
+    ori = - (score_size / 2) * total_stride
+    xx, yy = np.meshgrid([ori + total_stride * dx for dx in range(score_size)], [ori + total_stride * dy for dy in range(score_size)])
+    xx, yy = np.tile(xx.flatten(), (anchor_num, 1)).flatten(), np.tile(yy.flatten(), (anchor_num, 1)).flatten()
+    anchor[:, 0], anchor[:, 1] = xx.astype(np.float32), yy.astype(np.float32)
+
+    return anchor
+
+#parameters for another functions
+class TrackerConfig(object):
+    windowing = 'cosine'
+    exemplar_size = 127
+    instance_size = 271
+    total_stride = 8
+    score_size = (instance_size-exemplar_size)/total_stride+1
+    context_amount = 0.5
+    ratios = [0.33, 0.5, 1, 2, 3]
+    scales = [8, ]
+    anchor_num = len(ratios) * len(scales)
+    anchor = []
+    penalty_k = 0.055
+    window_influence = 0.42
+    lr = 0.295
+
+#initialization of network
+def SiamRPN_init(im, target_pos, target_sz, net, kernel_r1, kernel_cls1):
```

Comment:
Let'sre replace separate init and track functions to SiamRPN class with __init__ and track methods. So we will remove `state` dict and will just use class fields